controls/radvirtualgrid/mvvm.md

```diff
@@ -0,0 +1,327 @@
+                }
+            }
+        }
+
+        public int StadiumCapacity
+        {
+            get { return this.stadiumCapacity; }
+            set
+            {
+                if (value != this.stadiumCapacity)
+                {
+                    this.stadiumCapacity = value;
+                    this.OnPropertyChanged("StadiumCapacity");
+                }
+            }
+        }
+
+        public Club()
+        {
+
+        }
+
+        public Club(string name, DateTime established, int stadiumCapacity)
+        {
+            this.name = name;
+            this.established = established;
+            this.stadiumCapacity = stadiumCapacity;
+        }
+
+        protected virtual void OnPropertyChanged(PropertyChangedEventArgs args)
+        {
+            PropertyChangedEventHandler handler = this.PropertyChanged;
+            if (handler != null)
+            {
+                handler(this, args);
+            }
+        }
+
+        private void OnPropertyChanged(string propertyName)
+        {
+            this.OnPropertyChanged(new PropertyChangedEventArgs(propertyName));
+        }
+
+        public override string ToString()
+        {
+            return this.Name;
+        }
+
+        public static ObservableCollection<Club> GetClubs()
+        {
+            ObservableCollection<Club> clubs = new ObservableCollection<Club>();
+            Club club;
+            int id = 0;
+
+            // Liverpool
+            club = new Club("Liverpool", new DateTime(1892, 1, 1), 45362) { PrimaryKey = id};
+            clubs.Add(club);
+
+            id++;
+            // Manchester Utd.
+            club = new Club("Manchester Utd.", new DateTime(1878, 1, 1), 76212) {PrimaryKey = id};
+            clubs.Add(club);
+
+            id++;
+            // Chelsea
+            club = new Club("Chelsea", new DateTime(1905, 1, 1), 42055) { PrimaryKey = id };
+            clubs.Add(club);
+
+            id++;
+            // Arsenal
+            club = new Club("Arsenal", new DateTime(1886, 1, 1), 60355) { PrimaryKey = id };
+            clubs.Add(club);
+
+            return clubs;
+        }
+    }
+{{endregion}}
+
+## Implementing a custom DataProvider
+
+Then, a custom __DataProvider__ that handles some specific scenarios will be defined. Speaking shortly, it demonstrates how the selection of the control can be persisted when sorting or filtering. Note, that by default it is being cleared when processing these operations. Also, hiding a given column, defining a custom editor, setting a given column to be readonly and applying a custom header is demonstrated.
+
+#### __[C#] Example 2: Implementing a Custom DataProvider__
+
+{{region radvirtualgrid-mvvm_1}}
+	public class MyDataProvider : DataProvider
+	    {
+	        private List<int> selectedIndexes;
+	        public MyDataProvider(IEnumerable source) : base(source)
+	        {
+	            this.selectedIndexes = new List<int>();
+	        }
+	
+	        /// <summary>
```

Comment:
Do we need so much space to add the comment? I would recommend you to use simple comments "'\\" instead of the summary tag.